libkbfs/ongoing_work_limiter.go

```diff
@@ -0,0 +1,40 @@
+package libkbfs
+
+import "sync"
+
+// OngoingWorkLimiter limits maximum number of routines that can work on a same
+// type of thing at the same time. For example, it can be used for limiting
+// number of ongoing rekeys.
+type OngoingWorkLimiter struct {
+	cond     sync.Cond
```

Comment:
I think it might be more Go-like (and would integrate better with contexts) to use a fixed-capacity channel instead.  Then `Wait` could try to send on it, and `Done()` would drain from it.  Then you don't have to track capacity yourself, and get context cancellation without having to fork a goroutine (see comment below).